foam_controlwrapper/tests/test_controlwrapper.py

```diff
@@ -6,65 +6,306 @@
 """
 
 import unittest
-from foam_controlwrapper.foam_controlwrapper import FoamControlWrapper
+
+from simphony.cuds.mesh import Mesh, Face, Point, Cell
 from simphony.core.cuba import CUBA
+from simphony.core.data_container import DataContainer
+from simphony.io.h5_cuds import H5CUDS
+from foam_controlwrapper.foam_controlwrapper import FoamControlWrapper
 from foam_controlwrapper.cuba_extension import CUBAExt
-import shutil
-import logging
+from foam_controlwrapper.foam_files import FoamFiles
 import os
-from PyFoam.Execution.UtilityRunner import UtilityRunner
-
-logger = logging.getLogger(__name__)
+import shutil
 
 
 class FoamControlWrapperTestCase(unittest.TestCase):
     """Test case for FoamControlWrapper class"""
     def setUp(self):
-        pass
+        self.mesh = Mesh(name="mesh1")
+
+        self.points = [
+            Point(
+                (0.0, 0.0, 0.0)),
+            Point(
+                (1.0, 0.0, 0.0)),
+            Point(
+                (1.0, 1.0, 0.0)),
+            Point(
+                (0.0, 1.0, 0.0)),
+            Point(
+                (0.0, 0.0, 1.0)),
+            Point(
+                (1.0, 0.0, 1.0)),
+            Point(
+                (1.0, 1.0, 1.0)),
+            Point(
+                (0.0, 1.0, 1.0))
+        ]
+
+        puids = [self.mesh.add_point(point) for point in self.points]
+
+        self.faces = [
+            Face([puids[0], puids[3], puids[7], puids[4]],
+                 data=DataContainer({CUBA.LABEL: 0})),
+            Face([puids[1], puids[2], puids[6], puids[5]],
+                 data=DataContainer({CUBA.LABEL: 1})),
+            Face([puids[0], puids[1], puids[5], puids[4]],
+                 data=DataContainer({CUBA.LABEL: 2})),
+            Face([puids[3], puids[2], puids[6], puids[7]],
+                 data=DataContainer({CUBA.LABEL: 3})),
+            Face([puids[0], puids[1], puids[2], puids[3]],
+                 data=DataContainer({CUBA.LABEL: 4})),
+            Face([puids[4], puids[5], puids[6], puids[7]],
+                 data=DataContainer({CUBA.LABEL: 5}))
+
+        ]
+
+        [self.mesh.add_face(face) for face in self.faces]
+
+        self.cells = [
+            Cell(puids)
+        ]
+
+        self.puids = puids
+
+        [self.mesh.add_cell(cell) for cell in self.cells]
+
+    def tearDown(self):
+        FoamFiles().remove_parser_files(os.getcwd())
+
+    def test_add_mesh(self):
```

Comment:
To clarify, executing code and running workflows to detect if anything breaks is still a valid tests methodoligy (i.e. *smoke testing* ) but that check mainly that the code runs not that the code runs **correctly** and thus it is mainly used for higher level integration tests.